### ISSUE TITLE
[Bug]:  Results of (padding=True, pad_maps=False) are worse than  (padding=False, pad_maps=True) of EfficientAD 

### ISSUE THREAD

**Issue by jiamfeng** [openvinotoolkit/anomalib#2009]
### Describe the bug

When I trained EfficientAD on my own dataset, I found results of (padding=True, pad_maps=False) are worse than default setting, I am confused about this problem. results is below.
![image](https://github.com/openvinotoolkit/anomalib/assets/18417268/9d441962-de02-42a5-a726-a536cf036129)

### Dataset

Folder

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

# Create the datamodule
datamodule = Folder(
    name="white_point",
    root="/home/data/QiangBan/white_point/train",
    normal_dir="good",
    test_split_mode=TestSplitMode.SYNTHETIC
)
# datamodule = MVTec()

# Setup the datamodule

datamodule.setup()
model = EfficientAd(
    model_size=EfficientAdModelSize.S,
    #padding=True,
    #pad_maps=False
)

# prepare tiling configuration callback
#tiler_config_callback = TilerConfigurationCallback(enable=True, tile_size=[512, 512], stride=64)

# pass the tiling configuration callback to engine
engine = Engine(image_metrics=["AUROC"], pixel_metrics=["AUROC"], max_epochs=100)

# train the model (tiling is seamlessly utilized in the background)
engine.fit(datamodule=datamodule, model=model)

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [3.10.0]
- Anomalib version: [1.0.1 latest]
- PyTorch version: [2.0.0]
- CUDA/cuDNN version: [e.g. 11.8]
- GPU models and configuration: [1x GeForce RTX 3090]
- Any other relevant information: [I'm using a custom dataset]

### Expected behavior

Get the results of default configuration of model with 4 pixels pad is code.

### Pip/GitHub

GitHub

### What version/branch did you use?

latest

### Configuration YAML

```yaml
Not use yaml, use python file to train. codes is above.
```

### Logs

```shell
No
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by papago2355** [comment]
What about set both padding and pad_map to 'True'

**Comment by alexriedel1** [comment]
From an qualitative view, (padding=False, pad_maps=True) gives you worse results because it misses all the anomalies in the border region of your image. However I think anomalies are well detected in your custom data, the reason for the too large anomaly area might also lie in disadvantageous segmentation mask labeling

**Comment by jiamfeng** [comment]
> From an qualitative view, (padding=False, pad_maps=True) gives you worse results because it misses all the anomalies in the border region of your image. However I think anomalies are well detected in your custom data, the reason for the too large anomaly area might also lie in disadvantageous segmentation mask labeling

(padding=False, pad_maps=True) get the better results. 

> What about set both padding and pad_map to 'True'

I don't try this config

**Comment by jiamfeng** [comment]
> What about set both padding and pad_map to 'True'

I don't try this config, According to the results of padding =True, I think it can't be well detected.

**Comment by alexriedel1** [comment]
Im not so sure about:
> The result of (padding=False, pad_maps=True) is better. 

Because of the pad_maps you have some border regions of the image where anomalies are not detected. In your case this affects even a separate anomaly in the top right border for example
![image](https://github.com/openvinotoolkit/anomalib/assets/54716527/55d85fec-c657-4e1e-b797-31a3d9ae3ce4)

**Comment by jiamfeng** [comment]
> Im not so sure about:
> 
> > The result of (padding=False, pad_maps=True) is better.
> 
> Because of the pad_maps you have some border regions of the image where anomalies are not detected. In your case this affects even a separate anomaly in the top right border for example ![image](https://private-user-images.githubusercontent.com/54716527/325524663-55d85fec-c657-4e1e-b797-31a3d9ae3ce4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQwOTg1MTcsIm5iZiI6MTcxNDA5ODIxNywicGF0aCI6Ii81NDcxNjUyNy8zMjU1MjQ2NjMtNTVkODVmZWMtYzY1Ny00ZTFlLWI3OTctMzFhM2Q5YWUzY2U0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI2VDAyMjMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwZjAyNTVhNjA0MDZiNDRhM2I5NjkwOThjYWVmMzI4MjgwNjhiNDkyNjAyNDcxN2Q0MDM4YTRiYTgyZTc3Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.p08pMKg8NSEwnrKizIL4bELg4AMgSl7_MAEWHHkxWK4)

Yes, Even though the result of (padding=False, pad_maps=True), it will lead to more backgroud False Positive, I think may be the distillation pretrained model is important, I am not sure, may be need to do some experiments to prove.

**Comment by gitfri021** [comment]
I just trained a model with padding = true and pad_maps=false, the results are worse than the vice versa setting. The issue is that with the first setting, too much background is detected as anomalies and the score is also high.
The issue with the second setting is that it misses out anomalies in the edges of the image which is very crucial for my project.
How do we fix this problem?
